packages/components/src/color-picker/test/index.tsx

```diff
@@ -342,4 +342,85 @@ describe( 'ColorPicker', () => {
 			} );
 		} );
 	} );
+
+	describe.each( [
+		[ 'hsl', 'HSL', '75', '#ffffff12' ],
+		[ 'rgb', 'RGB', '75', '#ffffff12' ],
+	] )(
+		'Alpha-enabled %s format',
+		( format, formatLabel, alphaValue, expected ) => {
+			it( `should update alpha correctly when ${ formatLabel } format is selected`, async () => {
+				const user = userEvent.setup();
+				const onChange = jest.fn();
+
+				const ControlledColorPicker = ( {
+					onChange: onChangeProp,
+					...restProps
+				}: React.ComponentProps< typeof ColorPicker > ) => {
+					const [ colorState, setColorState ] =
+						useState( '#ffffff80' );
+
+					const internalOnChange: typeof onChangeProp = (
+						newColor
+					) => {
+						onChangeProp?.( newColor );
+						setColorState( newColor );
+					};
+
+					return (
+						<>
+							<ColorPicker
+								{ ...restProps }
+								onChange={ internalOnChange }
+								color={ colorState }
+							/>
+						</>
+					);
+				};
+
+				render(
+					<ControlledColorPicker onChange={ onChange } enableAlpha />
+				);
+
+				const formatSelector = screen.getByRole( 'combobox' );
+				expect( formatSelector ).toBeVisible();
+				await user.selectOptions( formatSelector, format );
+
+				const alphaInput = screen.getByRole( 'spinbutton', {
+					name: 'Alpha',
+				} );
+				expect( alphaInput ).toBeVisible();
+
+				const alphaSliders = screen.getAllByRole( 'slider', {
+					name: 'Alpha',
+				} );
+
+				const alphaSlider = alphaSliders.at( -1 )!;
+
+				expect( alphaSlider ).toHaveValue( '50' );
+				expect( alphaInput ).toHaveValue( Number( '50' ) );
```

Comment:
Personal preferences, but I'd prefer `50` instead of `Number( '50' )`. Here and everywhere else throughout the file